src/proxy/moralis.rs

```diff
@@ -0,0 +1,330 @@
+/// It also includes a `SignedMessage` for authentication and validation, confirming the legitimacy of the request and enhancing security.
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+pub(crate) struct MoralisPayload {
+    uri: Url,
+    pub(crate) signed_message: SignedMessage,
+}
+
+pub(crate) async fn proxy_moralis(
+    cfg: &AppConfig,
+    mut req: Request<Body>,
+    remote_addr: &SocketAddr,
+    payload: MoralisPayload,
+    x_forwarded_for: HeaderValue,
+    proxy_route: &ProxyRoute,
+) -> GenericResult<Response<Body>> {
+    if proxy_route.authorized {
+        if let Err(e) = insert_jwt_to_http_header(cfg, req.headers_mut()).await {
+            log::error!(
+                "{}",
+                log_format!(
+                    remote_addr.ip(),
+                    payload.signed_message.address,
+                    req.uri(),
+                    "Error inserting JWT into HTTP header: {}, returning 500.",
+                    e
+                )
+            );
+            return response_by_status(StatusCode::INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    let original_req_uri = req.uri().clone();
+
+    if let Err(e) = modify_request_uri(&mut req, &payload, proxy_route).await {
+        log::error!(
+            "{}",
+            log_format!(
+                remote_addr.ip(),
+                payload.signed_message.address,
+                original_req_uri,
+                "Error modifying request Uri: {}, returning 500.",
+                e
+            )
+        );
+        return response_by_status(StatusCode::INTERNAL_SERVER_ERROR);
+    }
+
+    let additional_headers = &[
+        header::CONTENT_LENGTH,
+        HeaderName::from_bytes(X_AUTH_PAYLOAD.as_bytes())?,
+    ];
+    remove_unnecessary_headers(&mut req, additional_headers)?;
+
+    req.headers_mut()
+        .insert(HeaderName::from_static(X_FORWARDED_FOR), x_forwarded_for);
+    req.headers_mut()
+        .insert(header::ACCEPT, APPLICATION_JSON.parse()?);
+
+    let https = HttpsConnector::new();
+    let client = hyper::Client::builder().build(https);
+
+    let target_uri = req.uri().clone();
+    let res = match client.request(req).await {
+        Ok(t) => t,
+        Err(e) => {
+            log::warn!(
+                "{}",
+                log_format!(
+                    remote_addr.ip(),
+                    payload.signed_message.address,
+                    original_req_uri,
+                    "Couldn't reach {}: {}. Returning 503.",
+                    target_uri,
+                    e
+                )
+            );
+            return response_by_status(StatusCode::SERVICE_UNAVAILABLE);
+        }
+    };
+
+    Ok(res)
+}
+
+/// Modifies the URI of an HTTP request by replacing it to outbound URI specified in `ProxyRoute`,
+/// while incorporating the path and query parameters from the payload's URI.
+async fn modify_request_uri(
+    req: &mut Request<Body>,
```

Comment:
From the other PR as I mentioned we should not convert requests while proxying them. Please consider changing the signed message handling in this repository to take it from `KOMODO_DEFI_SIGNED_MESSAGE` so we don't have to rely on request body to provide this data.

Reply:
This function doesnt rely on Body, it changes req's url to `outbound_route`. Currently signed message is parsed from req header for Moralis feature.

Reply:
>  Currently signed message is parsed from req header for Moralis feature

Check `generate_payload_from_req` function

Reply:
Oh, that's nice. But this should be done for all supported services including quicknode.

Reply:
> Oh, that's nice. But this should be done for all supported services including quicknode.

Hmm you mean provide signed message in header? That would be nice yeah, I will do it. But I would like to keep json-rpc data in Body since quicknode requires POST methods

Reply:
> Hmm you mean provide signed message in header? That would be nice yeah, I will do it. But I would like to keep json-rpc data in Body since quicknode requires POST methods

We only need signed messages to be included in header. Requests will remain the same, we just need to put a header on them. No need to edit RPC data in the body.